lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -35,6 +40,46 @@ def __init__(self, apis, concrete_artifacts_manager):
         self._apis = apis
         self._concrete_art_mgr = concrete_artifacts_manager
 
+    def _get_collection_versions_with_api_order(self, requirement, api_lookup_order):
+        # type: (Candidate, Iterable[GalaxyAPI]) -> Iterable[Tuple[GalaxyAPI, str]]
+        """Helper for get_collection_versions. Yield api, version pairs for all APIs and reraise the last error if no valid API was found."""
+        found_api = last_error = None
+
+        for api in api_lookup_order:
+            versions, error = self._get_collection_versions_with_api(requirement, api)
+
+            if error is None:
+                found_api = api
+                for version in versions:
+                    yield api, version
+            else:
+                last_error = error
+
+        if found_api is None and last_error is not None:
+            raise last_error
+
+    def _get_collection_versions_with_api(self, requirement, api):
+        # type: (Candidate, GalaxyAPI) -> Tuple[Optional[list], Optional[Exception]]
```

Comment:
It's not a candidate according to how it's being called.
```suggestion
        # type: (Requirement, GalaxyAPI) -> Tuple[Optional[list], Optional[Exception]]
```